docs/messaging-implementation.md

```diff
@@ -0,0 +1,168 @@
+# In-app messaging (technical implementation)
+
+All forms of in-app messages (notifications, mascot announcements, popup announcements, etc.) share a single data model. They depend on a
+JSON feed containing certain attributes and flags. See the [configuraton](configuration.md) doc for information about how to
+point uw-frame to your desired feed.
+
+## Example message
+
+*Note: All messages must be contained within the "messages" array.*
+
+```json
+{
+    "messages": [
+        {
+            "id": 1,
+            "title": "You need to modify your NetID account to activate essential UW Services.",
+            "titleShort": null,
+            "description": "Because this message's messageType is 'notification', this field is not actually needed.",
+            "descriptionShort": null,
+            "messageType": "notification",
```

Comment:
I still see promise in implying, deriving the messageType from semantic metadata. It's a notification because it's an actionable message about an external system.